### PR TITLE
Event queue threading

### PR DESCRIPTION
**Pull request by sebastiansimson**
Make add and flush operations async in a single thread executor

### CODE REVIEW

File: castle/src/main/java/io/castle/android/queue/EventQueue.java [hunk 1]
```diff
@@ -58,17 +64,27 @@ private synchronized File getFile(Context context) {
     }
 
     private synchronized void init(Context context) throws IOException {
+        executor = Executors.newSingleThreadExecutor();
```
Comment (opengamer29) [inline castle/src/main/java/io/castle/android/queue/EventQueue.java:67]: I think passing executor from Castle::setup will be a more flexible and pretty chip to implement, just one another Castle::setup method with executor in parameters.
But maybe it's redundant :)

Comment (sebastiansimson) [reply castle/src/main/java/io/castle/android/queue/EventQueue.java:67]: The reasoning I had for doing it this way is to make sure that there is always just one queue operation running concurrently. Being able to set the executor on your own would make it possible to provide a multi thread executor and it feels unnecessary and hard to test for our use case.

**Review by opengamer29 — APPROVED** [review]

**Comment by opengamer29** [comment]
@sebastiansimson any news about this MR?

**Comment by sebastiansimson** [comment]
Making sure that the automated tests work as expected then I will prepare a release with this PR included.

**Pull request merged**